drivers/counter/counter_nrfx_timer.c

```diff
@@ -131,11 +190,15 @@ static int counter_nrfx_set_alarm(struct device *dev, u8_t chan_id,
 	}
 
 	cc_val = counter_nrfx_get_cc_value(dev, alarm_cfg);
+	chdata->callback = alarm_cfg->callback;
+	chdata->user_data = alarm_cfg->user_data;
 
-	nrfx_config->ch_data[chan_id].callback = alarm_cfg->callback;
-	nrfx_config->ch_data[chan_id].user_data = alarm_cfg->user_data;
+	if (set_cc(dev, ID_TO_CC(chan_id), cc_val) != 0) {
+		counter_alarm_callback_t clbk = chdata->callback;
 
-	nrfx_timer_compare(timer, ID_TO_CC(chan_id), cc_val, true);
+		chdata->callback = NULL;
+		clbk(dev, chan_id, cc_val, chdata->user_data);
```

Comment:
As already pointed in #15510, this behavior should be configurable (BTW the commit message mentions only reporting).